### ISSUE TITLE
Update docs to remove image

### ISSUE THREAD

**Issue by wielandbrendel** [bethgelab/foolbox#12]
In the documentation we quite frequently refer to "images" but adversarials can be generated for any type of input. We should thus replace "images" with a more generic term like "input".

**Comment by jonasrauber** [comment]
Along with that we should analyze where the code is image-specific. For that it would be great to have an actual use-case.

**Comment by AngusG** [comment]
I found [TensorFlowModel](http://foolbox.readthedocs.io/en/latest/modules/models.html#foolbox.models.TensorFlowModel) to be image/cnn specific. I wanted to import a dense net for MNIST w/ input shape (batch_size, 784) but it ended up being easier to just use a conv architechture.

**Comment by wielandbrendel** [comment]
@AngusG Can you post a minimal working example of your problem (in a new issue)? Thanks!

**Comment by jonasrauber** [comment]
@wielandbrendel We certainly have to rename the parameters and move image-specific ones (e.g. channel_axis) to the attacks that need it along with refactoring the preprocessing (e.g. reversing axes like channel order) and adding support for different prediction types / gradients / losses. I plan to work on this next month.

@AngusG Nevertheless, I don't see why a simple dense net shouldn't work the way it is right now. For me, this works:

```python3
import foolbox
import tensorflow as tf
import numpy as np

inputs = tf.placeholder(tf.float32, shape=(None, 784))
logits = tf.layers.dense(inputs, 10)

init_op = tf.global_variables_initializer()

np.random.seed(2)
W = np.random.rand(784, 10).astype(np.float32)

assign_op = tf.assign(tf.global_variables()[0], W)

np.random.seed(22)
example_input = np.random.rand(784)

with foolbox.models.TensorFlowModel(inputs, logits, (0, 1)) as model:
    attack = foolbox.attacks.FGSM(model)

    model.session.run(init_op)
    model.session.run(assign_op)  # replace with restorer

    example_label = np.argmax(model.predictions(example_input))
    print(example_label)

    adversarial = attack(example_input, example_label, unpack=False)
    print(np.argmax(model.predictions(adversarial.image)))

    print(adversarial.distance)
```

outputs

```
7
9
rel. MSE = 0.01427  %
```

**Comment by AngusG** [comment]
Thanks for the example @jonasrauber. The wrapper is not cnn specific as I suspected, although I'm unsure as to why the default setting of `channel_axis=3` works for input data with less than 3 axis. This also implies to me that the wrapper accepts data of shape `[batch_size, h, w, channels]` but a batch dimension of one is added automatically in models/base -- predictions(self, image).
```
from tensorflow.examples.tutorials.mnist import input_data
mnist = input_data.read_data_sets(<data_dir>, dtype=tf.float32, one_hot=True)
image, label = mnist.test.next_batch(batch_size)
```
`image` has shape `(batch_size, 784)` here.
Reshaping to `image.reshape(784)` works (for a batch_size of 1). A simple check to see if there is already a batch dimension would be a nice. I don't know why I was looking for a workaround before where I could leave the shape as-is. 

What would be involved to enable `batch_predictions`? This is something I might be interested in contributing.

**Comment by jonasrauber** [comment]
As you already figured out yourself, `batch_predictions` is already supported: https://github.com/bethgelab/foolbox/issues/57

Regarding the `channel_axis`: the default setting to 3 doesn't produce an error even if there is no such dimension because the channel_axis is only used by some image-specific attacks (only for those, you will see an error); in the future, we will remove `channel_axis` and other image-specific code / parameter names …
